### PR TITLE
Updates several packages for tgui

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
Woah! Shiny!

Updates most of tgui's internals. Nothing player facing. Tested and working
## Why It's Good For The Game
Good housekeeping
## Changelog
N/A nothing player facing

### CODE REVIEW

File: tgui/babel.config.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const createBabelConfig = (options) => {
       ...presets,
     ].filter(Boolean),
     plugins: [
-      [require.resolve('@babel/plugin-proposal-class-properties'), {
+      [require.resolve('@babel/plugin-transform-class-properties'), {
```
Comment (jlsnow301) [inline tgui/babel.config.js:26]: proposal is deprecated

> This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.

https://www.npmjs.com/package/@babel/plugin-proposal-class-properties

File: tgui/.eslintrc.yml [hunk 2]
```diff
@@ -651,7 +651,7 @@ rules:
   ## Enforce ES5 or ES6 class for React Components
   react/prefer-es6-class: error
   ## Enforce that props are read-only
-  react/prefer-read-only-props: error
+  # react/prefer-read-only-props: error
```
Comment (jlsnow301) [inline tgui/.eslintrc.yml:654]: we don't do this anywhere and with the newer versioning it's detecting it and attempting to autofix. it does this terribly by appending `readonly` to every type, which could instead be wrapped by `Readonly<x>`

**Comment by MrStonedOne** [comment]
```
➤ YN0002: │ babel-plugin-inferno@npm:6.7.0 doesn't provide @babel/core (peb942), requested by @babel/plugin-syntax-jsx
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash>
is the six-letter p-prefixed code
➤ YN0000: └ Completed in 0s 498ms
```
yarn is complaining about something.

**Comment by jlsnow301** [comment]
> yarn is complaining about something, seems to be a warning as exit code was still `0`

this is now resolved. unfortunately babel-plugin-inferno is no longer maintained, I think a workaround is the only solution here

**Pull request merged**